data/src/main/java/com/teamagam/gimelgimel/data/message/adapters/MessageDataMapper.java

```diff
@@ -129,23 +140,56 @@ public void visit(MessageImageData message) {
 
         @Override
         public void visit(MessageSensorData message) {
-            SensorMetadata sensorMetadata = convertSensorMetaData(message.getContent());
+            SensorMetadata sensorMetadata = convertContent(message.getContent());
             mMessage = new MessageSensor(
                     message.getMessageId(),
                     message.getSenderId(),
                     message.getCreatedAt(),
                     sensorMetadata);
         }
 
-        private SensorMetadata convertSensorMetaData(SensorMetadataData sensorMetadataData) {
+        @Override
+        public void visit(MessageUserLocationData message) {
+            LocationSample convertedLocationSample = mLocationSampleAdapter.transform(
+                    message.getContent());
+            mMessage = new MessageUserLocation(message.getMessageId(),
+                    message.getSenderId(), message.getCreatedAt(),
+                    convertedLocationSample);
+        }
+
+        @Override
+        public void visit(MessageVectorLayerData message) {
+            VectorLayer vl = convertContent(message.getContent());
+            mMessage = new MessageVectorLayer(message.getMessageId(),
+                    message.getSenderId(),
+                    message.getCreatedAt(),
+                    vl);
+        }
+
+        private VectorLayer convertContent(VectorLayerData content) {
+            return new VectorLayer(content.getId(), content.getName(),
+                    tryParseUrl(content));
+        }
+
+        private URL tryParseUrl(VectorLayerData content) {
```

Comment:
Why not use a parameter of type String instead of VectorLayerData and call the method with content.getRemoteUrl(), as in the other params that are sent to the VectorLayer constructor?